src/plugins/data_view_editor/public/components/confirm_modals/edit_data_view_changed_modal.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState } from 'react';
+import { EuiCallOut, EuiSpacer, EuiConfirmModal, EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+const geti18nTexts = (dataViewName?: string) => ({
+  cancelButtonText: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.cancelButtonLabel',
+    {
+      defaultMessage: 'Cancel',
+    }
+  ),
+  confirmButtonText: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.saveButtonLabel',
+    {
+      defaultMessage: 'Save',
+    }
+  ),
+  warningEditingDataView: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.warningEditingDataView',
+    {
+      defaultMessage:
+        'Are you sure you want to edit this data view? Anything that depends upon this might not work as expected.',
+    }
+  ),
+  typeConfirm: i18n.translate('indexPatternEditor.editDataView.editConfirmationModal.typeConfirm', {
+    defaultMessage: 'Enter CHANGE to continue',
+  }),
+  titleConfirmChanges: i18n.translate(
+    'indexPatternEditor.editDataView.editConfirmationModal.title',
+    {
+      defaultMessage: `Edit {name}`,
+      values: {
+        name: dataViewName ? `'${dataViewName}'` : 'data view',
+      },
+    }
+  ),
+});
+
+interface Props {
+  dataViewName?: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+}
+
+export const EditDataViewChangedModal: React.FC<Props> = ({
+  dataViewName,
+  onCancel,
+  onConfirm,
+}) => {
+  const i18nTexts = geti18nTexts(dataViewName);
+  const [confirmContent, setConfirmContent] = useState<string>('');
+
+  return (
+    <EuiConfirmModal
+      title={i18nTexts.titleConfirmChanges}
+      data-test-subj="runtimeFieldSaveConfirmModal"
+      cancelButtonText={i18nTexts.cancelButtonText}
+      confirmButtonText={i18nTexts.confirmButtonText}
+      confirmButtonDisabled={confirmContent?.toUpperCase() !== 'CHANGE'}
```

Comment:
This will break when a user-facing label is translated. I think we should change the translation label to accept value which would be a const string `CHANGE` 

```
typeConfirm: i18n.translate('indexPatternEditor.editDataView.editConfirmationModal.typeConfirm', {
     defaultMessage: 'Enter {changeKeyword} to continue',
     values: { changeKeyword: 'CHANGE' }
   }),
```

Reply:
the modal has been changed to a confirm modal after discussions.